divergence_detector.py

```
@@ -592,6 +604,122 @@ def get_divergence_trading_signals(self, divergences: Dict) -> Dict:
+                else:
+                    return 'STRONG_BULL'
+            elif price_change <= self.bear_market_threshold:
+                return 'BEAR'
+            else:
+                return 'NEUTRAL'
+                
+        except Exception as e:
+            logger.error(f"Error analyzing market regime: {e}")
+            return 'NEUTRAL'
+    
+    def _apply_trend_filter(self, divergence_result: Dict, market_regime: str, divergence_type: str) -> Dict:
```

Ruff (0.12.2)

Unused method argument: `divergence_type`

(ARG002)